src/checked_transaction.rs

```diff
@@ -92,6 +104,58 @@ impl CheckedTransaction {
         self.checked_signatures
     }
 
+    pub const fn metadata(&self) -> Option<&Metadata> {
+        self.transaction.metadata()
+    }
+
+    /// Update change and variable outputs.
+    ///
+    /// `revert` will signal if the execution was reverted. It will refund the unused gas cost to
+    /// the base asset and reset output changes to their initial balances.
+    ///
+    /// `unused_gas` will be added to the output that contains the base asset
+    ///
+    /// `balances` will contain the current state of the free balances
+    pub fn update_outputs<I>(&mut self, revert: bool, unused_gas: Word, balances: &I)
```

Comment:
I'm thinking we should treat unused_gas as gas instead of a refund amount. That way we can consolidate all the logic involving gas_price_factor into fuel-tx, and fuel-vm just needs to report how much GGAS is left to this method.

Reply:
That's a good suggestion!

Updated in https://github.com/FuelLabs/fuel-tx/pull/162/commits/8c5b71f0fc098e80ea5e0479301299785e37c04f